runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,758 @@
+<
+The fourth, optional, argument is a table with keys that modify the behavior
+of the mapping such as those from `:map-arguments`. The following are the most
+useful options:
+• `buffer`: If given, only set the mapping for the buffer with the specified
+  number; `0` or `true` means the current buffer. >lua
+    -- set mapping for the current buffer
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action, { buffer = true})
+    -- set mapping for the buffer number 4
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action, { buffer = 4})
+<
+• `silent`: If set to `true`, suppress output such as error messages. >lua
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action, { silent = true})
+<
+• `expr`: If set to `true`, do not execute the {rhs} but use the return value
+  as input. Special |keycodes| are converted automatically. For example, the following
+  mapping replaces <down> with <c-n> in the popupmenu only: >lua
+    vim.keymap.set('c', '<down>', function()
+      if vim.fn.pumvisible() == 1 then return '<c-n>' end
+      return '<down>'
+    end, { expr = true })
+<
+• `desc`: A string that is shown when listing mappings with, e.g., |:map|.
+  This is useful since Lua functions as {rhs} are otherwise only listed as
+  `Lua: <number> <source file>:<line>`. Plugins should therefore always use this
+  for mappings they create. >lua
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action,
+      { desc = 'Execute action from plugin' })
+<
+• `remap`: By default, all mappings are nonrecursive by default (i.e.,
+  |vim.keymap.set()| behaves like |:noremap|). If the {rhs} is itself a mapping
+  that should be executed, set `remap = true`: >lua
+    vim.keymap.set('n', '<Leader>ex1', '<cmd>echo "Example 1"<cr>')
+    -- add a shorter mapping
+    vim.keymap.set('n', 'e', '<Leader>ex1', { remap = true })
+<
+  The exception is if the {rhs} is a special |<Plug>| mapping, for which
+  `remap = true` by default: >lua
+    vim.keymap.set('n', '[%', '<Plug>(MatchitNormalMultiBackward)')
+<
+------------------------------------------------------------------------------
+Removing mappings                                       *lua-guide-mappings-del*
+
+A specific mapping can be removed with |vim.keymap.del()|:
+>lua
+    vim.keymap.del('n', '<Leader>ex1')
+    vim.keymap.del({'n', 'c'}, '<Leader>ex2', {buffer = true})
+<
+------------------------------------------------------------------------------
+See also:
+• `vim.api.`|nvim_get_keymap()|     - return all global mapping
+• `vim.api.`|nvim_buf_get_keymap()| - return all mappings for buffer
+
+==============================================================================
+Autocommands                                            *lua-guide-autocommands*
+
+An |autocommand| is a Vim command or a Lua (or Vimscript) function that is
+executed whenever one or more |events| are triggered automatically in response
```

Comment:
Hmm, bit of a wonky phrasing. It makes it sound like the trigger is automatic and not the command/function itself?

Reply:
oops, another rebase pileup. I'll fix!